app/assets/javascripts/admin/user/experience_modal_view.js

```diff
@@ -2,152 +2,433 @@
-          });
+  loadSingleUserEntries = (users: Array<APIUserType>): Array<SingleUserTableEntry> =>
+    users.length === 1
+      ? _.map(users[0].experiences, (value, domain) => ({
+          domain,
+          value,
+        })).sort(
+          Utils.localeCompareBy(([]: Array<SingleUserTableEntry>), entry =>
+            entry.domain.toLowerCase(),
+          ),
+        )
+      : [];
 
-          return this.sendUserToServer(newUser, user);
-        }
-        return Promise.resolve(user);
+  updatedAllUsers = async () => {
+    let newUserPromises: Array<Promise<APIUserType>>;
+    if (this.state.showOnlySharedExperiences && this.props.selectedUsers.length > 1) {
+      const relevantEntries = this.state.multipleUsersEntries.filter(entry => entry.isShared);
+      newUserPromises = this.props.selectedUsers.map(user => {
+        const newExperiences = { ...user.experiences };
+        relevantEntries.forEach(entry => {
+          newExperiences[entry.domain] = entry.value;
+        });
+        const newUser = { ...user, experiences: newExperiences };
+        return this.sendUserToServer(newUser, user);
+      });
+    } else {
+      const newExperiences = {};
+      if (this.props.selectedUsers.length === 1) {
+        this.state.singleUsersEntries.forEach(entry => {
+          newExperiences[entry.domain] = entry.value;
+        });
+      } else {
+        this.state.multipleUsersEntries.forEach(entry => {
+          newExperiences[entry.domain] = entry.value;
+        });
+      }
+      newUserPromises = this.props.selectedUsers.map(user => {
+        const newUser = { ...user, experiences: newExperiences };
+        return this.sendUserToServer(newUser, user);
       });
-
-      this.closeModal(newUserPromises);
     }
+    this.resolvePromisesAndCloseModal(newUserPromises);
   };
 
-  /**
-   * Save a user object to the server using an API call.
-   * @param newUser - A modified user object intended to be saved.
-   * @param oldUser - The original user object of `newUser`. Returned in case API call fails
-   *
-   */
   sendUserToServer(newUser: APIUserType, oldUser: APIUserType): Promise<APIUserType> {
     return updateUser(newUser).then(() => Promise.resolve(newUser), () => Promise.reject(oldUser));
   }
 
-  closeModal(usersPromises: Array<Promise<APIUserType>>): void {
+  resolvePromisesAndCloseModal(usersPromises: Array<Promise<APIUserType>>): void {
     Promise.all(usersPromises).then(
       newUsers => {
         this.setState({
-          domain: null,
-          level: null,
+          multipleUsersEntries: [],
+          singleUsersEntries: [],
         });
         this.props.onChange(newUsers);
       },
-      () => {
-        // do nothing and keep modal open
+      error => {
+        handleGenericError(error);
       },
     );
   }
 
-  render() {
-    if (!this.props.visible) {
-      return null;
+  recordModifiedAndExistedBefore = (record: SingleUserTableEntry): boolean => {
```

Comment:
I would rename it to `isRecordRevertable`. In general, it's a good best practice to name booleans like a question. At first,  I assumed the method would "record" something (since I thought record was a verb here :)).